File: src/main/java/com/indiepost/repository/ClickRepository.java
package com.indiepost.repository;

import com.indiepost.model.analytics.Click;
import org.springframework.data.repository.query.Param;

/**
 * Created by jake on 8/9/17.
 */
public interface ClickRepository {
    void save(Click click);

    Long countByLinkId(Long linkId);

    Long countValidClicksByLinkId(@Param("linkId") Long linkId);

    Long countAllClicksByCampaignId(Long campaignId);

    Long countValidClicksByCampaignId(Long campaignId);
}

File: src/main/kotlin/com/indiepost/repository/ClickRepository.kt
package com.indiepost.repository

import com.indiepost.model.analytics.Click
import org.springframework.data.repository.query.Param

/**
 * Created by jake on 8/9/17.
 */
interface ClickRepository {
    fun save(click: Click)

    fun countByLinkId(linkId: Long): Long

    fun countValidClicksByLinkId(@Param("linkId") linkId: Long): Long

    fun countAllClicksByCampaignId(campaignId: Long): Long

    fun countValidClicksByCampaignId(campaignId: Long): Long
}
